src/core/on-unhandled-rejection.js

```diff
@@ -0,0 +1,25 @@
+import { test } from "../test";
+
+import config from "./config";
+import { extend } from "./utilities";
+import { sourceFromStacktrace } from "./stacktrace";
+
+// Handle an unhandled rejection
+export default function onUnhandledRejection( reason ) {
```

Comment:
Is `reason` just an `Error` instance?

Reply:
No, `reason` is whatever the rejection value is.  An `Error` instance is almost certainly the most common thing that `reason` would be, but it could also be a failed XHR, a string, or nothing at all.

See small [demo JSBin](https://jsbin.com/muxahuw/edit?js,console,output) with the following content:

```js
window.addEventListener('unhandledrejection', function(event) {
  console.log('reason:', event.reason);
})

Promise.reject();
Promise.reject(null);
Promise.reject("asdf");
Promise.resolve().then(() => {
  throw new Error('boom');
})
```

Reply:
Ah okay, makes sense. Early-morning me couldn't quite make that connection.